statistics/handle/statscache.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handle
+
+import (
+	"encoding/binary"
+	"sync"
+
+	"github.com/pingcap/tidb/statistics"
+	"github.com/pingcap/tidb/util/kvcache"
+	"github.com/pingcap/tidb/util/memory"
+)
+
+// statsCache caches table statistics.
+type statsCache struct {
```

Comment:
Why don't we keep using the copy-on-write approach for stats cache?

Reply:
lock in cache can not be copy, but i don't know if there are other ways to copy lrucache ?

Reply:
> lock in cache can not be copy

Can you elaborate more on this? We'd better figure out an approach to read/write the list of LRUCache concurrently with high efficiency, otherwise this mechanism may incur significant performance regression when there are a lot of connections?